### PR TITLE
Switch from docopt to cligen for cli

### PR DESCRIPTION
**Pull request by SolitudeSF**
this drastically reduces binary size and startup time.
also bump required nim version.

### CODE REVIEW

File: .travis_setup.sh [hunk 1]
```diff
@@ -21,8 +21,8 @@ fi
 
 if [ ! -f "$HOME/.asdf/installs/nim/v0.19.0/bin/nim" ]; then
     echo "Installing nim v0.19.0 via asdf"
-    asdf install nim v0.19.0
+    asdf install nim v0.20.0
 fi
 
 echo "Setting nim v0.19.0 as global"
```
Comment (OldhamMade) [inline .travis_setup.sh]: Please update the output messages also. 😃

File: .travis_setup.sh [hunk 2]
```diff
@@ -21,8 +21,8 @@ fi
 
 if [ ! -f "$HOME/.asdf/installs/nim/v0.19.0/bin/nim" ]; then
```
Comment (OldhamMade) [inline .travis_setup.sh]: These checks need to be updated to reflect the version bump.

Comment (SolitudeSF) [reply .travis_setup.sh]: i might be blind, but thats unconfirmed

Comment (OldhamMade) [reply .travis_setup.sh]: Thanks for the updates. This line is linked to the caching set-up in travis; generally, the nim binary is cached between builds, making the tests run faster and reducing overhead on travis' platform. There's a catch where version bumps require clearing out the cache, but that's a minor inconvenience.

**Pull request merged**